lib/sufia/export/generic_file_converter.rb

```diff
@@ -0,0 +1,52 @@
+module Sufia
+  module Export
+    class GenericFileConverter
+      attr_accessor :id, :label, :depositor, :arkivo_checksum, :relative_path,
+                    :import_url, :resource_type, :title, :creator, :contributor,
+                    :description, :tag, :rights, :publisher, :date_created, :date_uploaded,
+                    :date_modified, :subject, :language, :identifier, :based_near,
+                    :related_url, :batch_id, :visibility, :versions
+      attr_accessor :bibliographic_citation, :source
+      attr_accessor :permissions
+
+      def initialize(gf)
+        @id = gf.id
+        @label = gf.label
+        @depositor = gf.depositor
+        @arkivo_checksum = gf.arkivo_checksum
+        @relative_path = gf.relative_path
+        @import_url = gf.import_url
+        @resource_type = gf.resource_type
+        @title = gf.title
+        @creator = gf.creator
+        @contributor = gf.contributor
+        @description = gf.contributor
+        @tag = gf.tag
+        @rights = gf.rights
+        @publisher = gf.publisher
+        @date_created = gf.date_created
+        @date_uploaded = gf.date_uploaded
+        @date_modified = gf.date_modified
+        @subject = gf.subject
+        @language = gf.language
+        @identifier = gf.identifier
+        @based_near = gf.based_near
+        @related_url = gf.related_url
+        @bibliographic_citation = gf.bibliographic_citation
+        @source = gf.source
```

Comment:
I tried a few things that "Should work with delegations and to_json.
Unfortunately they did not work, but if they did it would also make the to_json method really ugly.  Not sure we get an advantage in changing these to delegation